presto-main/src/main/java/com/facebook/presto/operator/scalar/QuantileDigestFunctions.java

```diff
@@ -63,6 +68,41 @@ public static long valueAtQuantileBigint(@SqlType("qdigest(bigint)") Slice input
         return new QuantileDigest(input).getQuantile(quantile);
     }
 
+    @ScalarFunction("quantile_at_value")
+    @Description("Given an input x between min/max values of qdigest, find which quantile is represented by that value")
+    @SqlType(StandardTypes.DOUBLE)
+    @SqlNullable
+    public static Double quantileAtValueDouble(@SqlType("qdigest(double)") Slice input, @SqlType(StandardTypes.DOUBLE) double value)
+    {
+        return quantileAtValueBigint(input, doubleToSortableLong(value));
+    }
+
+    @ScalarFunction("quantile_at_value")
+    @Description("Given an input x between min/max values of qdigest, find which quantile is represented by that value")
+    @SqlType(StandardTypes.DOUBLE)
+    @SqlNullable
+    public static Double quantileAtValueReal(@SqlType("qdigest(real)") Slice input, @SqlType(StandardTypes.REAL) long value)
+    {
+        return quantileAtValueBigint(input, floatToSortableInt(intBitsToFloat((int) value)));
+    }
+
+    @ScalarFunction("quantile_at_value")
+    @Description("Given an input x between min/max values of qdigest, find which quantile is represented by that value")
+    @SqlType(StandardTypes.DOUBLE)
+    @SqlNullable
+    public static Double quantileAtValueBigint(@SqlType("qdigest(bigint)") Slice input, @SqlType(StandardTypes.BIGINT) long value)
+    {
+        QuantileDigest digest = new QuantileDigest(input);
+        if (digest.getCount() == 0) {
```

Comment:
I think this should be treated like a value outside of the range of the qdigest (null), because really this is just a special case of the below check.  Actually, we could just remove this first check, because when it's empty the digest's max value is `Long.MIN_VALUE` and its min value is `Long.MAX_VALUE`.

Reply:
Sure, I agree returning null for empty digest, and merged into one if-block check.